subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -65,6 +65,38 @@ int dfu_ctx_mcuboot_set_b1_file(char *file, bool s0_active, char **update)
 	return 0;
 }
 
+#define MODULE "dfu"
+#define FILE_FLASH_IMG "mcuboot/flash_img"
+static int store_offset(void)
+{
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		char key[] = MODULE "/" FILE_FLASH_IMG;
+		int err = settings_save_one(key, &flash_img, sizeof(flash_img));
+
+		if (err) {
+			LOG_ERR("Problem storing offset (err %d)", err);
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static int settings_set(const char *key, size_t len_rd,
```

Comment:
This is called in the background when `settings_load` is called? (is it called any other times?) It populates the `flash_img` variable? Function docs :).

Reply:
Just followed the documentation example. I would assume it is populated by settings_load function since it's connected to the read_cb